lib/msal-node/src/crypto/PkceGenerator.ts

```diff
@@ -0,0 +1,67 @@
+/*
+ * Copyright (c) Microsoft Corporation. All rights reserved.
+ * Licensed under the MIT License.
+ */
+
+import { PkceCodes } from '@azure/msal-common';
+import { PKCEConstants } from './../utils/NodeConstants';
+import { Base64Encode } from './../encode/Base64Encode';
+const crypto = require('crypto');
+
+/**
+ * https://tools.ietf.org/html/rfc7636#page-8
+ */
+export class PkceGenerator {
+    /**
+     * generates the codeVerfier and the challenge from the codeVerfier
+     * reference: https://tools.ietf.org/html/rfc7636#section-4.1 and https://tools.ietf.org/html/rfc7636#section-4.2
+     */
+    async generatePkceCodes(): Promise<PkceCodes> {
```

Comment:
Can these be static (to be consistent with other classes)?

Reply:
well, honestly node doesn't need it `async` but my hands are tied as the interface defines this as `async`. I can take up a separate PR to change the interface which would include changes in `msal-browser`

Reply:
You can define these synchronously here if you want and return a Promise-wrapped value in the interface implementation above. making these static isn't affected by the interface.

Reply:
`static` and `async` cannot be used together. This implementation looked cleaner than wrapping promise return later.

Reply:
> static and async cannot be used together

Are you sure? Pretty sure they can.

Reply:
Did not compile for me. I can check in a non vscode environment.

Reply:
you can make them private static methods and call them the same way inside your public method.

Reply:
yeah! Well it is either that just not have static. Either way there is one unneeded qualifier on some functions. Not changing for now. Let us revisit, will make a note. Thanks for the feedback.